### PR TITLE
New DataCollection model for Portal client-side

### PR DESCRIPTION
**Pull request by dnahodil**
**Not ready for merging**

This data model change is to support multi-WMS, and later multi-WFS, data collections.

The big change is that previously the Portal used either GeoNetworkRecords or Layers and it's main data model. Now the main data model is a DataCollection. Most UI components will work directly with a DataCollection. DataCollections will link to a MetadataRecord (formerly a GeoNetworkRecord) and one or more Layers.

### CODE REVIEW

**Comment by dnahodil** [comment]
@markhepburn FYI these are the data model changes I was telling you about. This branch is still in development and will still change.

**Comment by dnahodil** [comment]
@jkburges this will still need some git rebase lovin'
We can talk about that tomorrow.

**Comment by dnahodil** [comment]
Bug spotted:
If you add a collection, then draw a bbox on step 2, and then add a second collection then bbox will disappear from the map. The constraint still appears to be applied correctly.

**Comment by dnahodil** [comment]
It seems we've reintroduced #1796 as well. The fix for that was to only load one data collection from a URL. I think loading many layers for one data collection (which will now be allowed) has the same effect.

**Comment by dnahodil** [comment]
:bug: I think I've broken ncWMS filtering.

**Comment by dnahodil** [comment]
:bug: Style panel doesn't update as you change selected layers

**Comment by jkburges** [comment]
Selecting an accordion folder should bring the associated layer to the top of the map, but it doesn't.

**Comment by jkburges** [comment]
Selecting an accordion folder should bring the associated layer to the top of the map, but it doesn't.

**Comment by jkburges** [comment]
> Selecting an accordion folder should bring the associated layer to the top of the map, but it doesn't.

Imma work on this one now.

**Comment by jkburges** [comment]
The time picker only affects the initially selected layer (rather than the currently selected layer).  Imma fix that.

**Comment by dnahodil** [comment]
Known to need tests:
DataCollection
DataCollectionStore
DataCollectionLayers
...

**Comment by dnahodil** [comment]
:bug: Clear and reset link on step 3 doesn't work

File: web-app/js/portal/data/DataCollectionStore.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        });
+    },
+
+    _recordLoaded: function(dataCollection) {
+        dataCollection.loaded = true;
+        Ext.MsgBus.publish(PORTAL_EVENTS.DATA_COLLECTION_ADDED, dataCollection);
+    },
+
+    _removeFromLayerStore: function(dataCollection) {
+        this.layerStore.removeUsingOpenLayer(dataCollection.getSelectedLayer());
```
Comment (dnahodil) [inline web-app/js/portal/data/DataCollectionStore.js]: Check this. It'll need to remove all layers from the layer store, I think. Not just the current one. Unless we actually add/remove them as we change them. (Possibly we do in DataCollectionLayers or whatever it is called now)

**Comment by dnahodil** [comment]
We need to confirm GoogleAnalytics tracking.
* For layer switcher
* Change to use DataCollection name rather than layer name
* Check that tracking is happening for changing scale range and style choice

**Pull request merged**